### ISSUE TITLE
JFrog Artifactory (repo.jenkins-ci.org) maintenance: Cloud migration from GCP to AWS by JFrog teams

### ISSUE THREAD

**Issue by dduportal** [jenkins-infra/helpdesk#3288]
## TL;DR;

The Sunday 18 December of 2022, JFrog teams will proceed to migrating our Artifactory instance `repo.jenkins-ci.org` (referenced as `jenkinsci.jfrog.io` for them) from their current GCP cloud to their "global platform" on AWS cloud.

## Why

The goal of this migration is to ensure that:

- The Jenkins Project sponsorship by JFrog is referenced and followed internally, as most of the initial person involved in this sponsorship are gone and that our instance was a kind of "forgotten" system

- Everyone from the Jenkins Infra team and governance board are able to open issues for the JFrog support (we are only 2-3 person who must use email: it is tedious and hard to track for their support)

- The instance benefits from the latest Artifactory updates and the whole JFrog platform monitoring (metrics, logs collection, customer portal)

- JFrog can keep maintaining our sponsorship at an acceptable cost for them

## What / How

- Migration of artefacts really started the 7 of December: JFrog team is running a (low-priority) copy of our artifacts from the current instance to the new one.
  - No impact for Jenkins project 
  - Multiple sync are planned until the real migration to retrieve the diff from the initial copy

- The operation planned for the 18 implies to migrate the database. 
  - To limit risks, the current instance will be shut down this day, right before the database migration.
  - Jenkins Infrastructure team will set ci.jenkins.io in "maintenance" mode as suggested by @daniel-beck : builds will be added in the queue without being processed (until the maintenance is finished). The goal is to avoid loosing build requests from contributors.
  - Sync of artefacts, TLS certificates (ref. #3259) and all other elements will be synced during the downtime, one last time, properly.
- Then the new instance will be started and the DNS `jenkinsci.jfrog.io` updated to this new instance

## Notes / Actions Item ⚠️ 

- Announce of the downtime to be done:
  - [x] Blog post on jenkins.io - https://www.jenkins.io/blog/2022/12/09/jfrog-maintenance/
  - [x] status.jenkins.io
  - [x] mailing list dev
  - [x] github organization pages for `jenkinsci` and `jenkins-infra`
- Allow the new instance IP to reach the LDAP
  - [x] Get the egress public IPs of the new instance from JFrog
  - [x] Add these IP in the allow list of the Service LB in front of the ldap pod
- Ensure that we have an external backup of the "upstream-gone" mirror repositories:
  - [ ] `javanet2`
- JFrog will keep the old instance for 1 month after the migration in case we have any issue that need a revert (or to retrieve files from the
- Action to run right before the operation (30 min. before):
  - Update status.jenkins.io (announce -> downtime)
  - Put ci.jenkins.io and trusted.ci.jenkins.io to "maintenance mode"
  - Send email + IRC message

**Comment by dduportal** [comment]
- Blog post PR at https://github.com/jenkins-infra/jenkins.io/pull/5769
- Initial announcment on status.jenkins.io (PR): https://github.com/jenkins-infra/status/pull/218

**Comment by dduportal** [comment]
Adding suggestion about ci.jenkins.io in maintenance mode as per https://groups.google.com/g/jenkinsci-dev/c/oNjz5afwwdA/m/vs-7x3r1GwAJ

- Updated issue description
- Updated status.jenkins.io: https://github.com/jenkins-infra/status/compare/main...dduportal:status:update/jfrog-maintenance?expand=1

**Comment by dduportal** [comment]
Public IPs are for the JFrog AWS `us-east-1`  region in https://jfrog.com/knowledge-base/what-are-artifactory-cloud-nated-ips/ as per JFrog feedback (https://groups.google.com/g/jenkins-infra-team/c/CPdUSBLp_BM/m/EJ_wXVvqBgAJ => private mailing list)

**Comment by dduportal** [comment]
- Time windows had been communicated: starting at 09:00am UTC, for 1 to 6 hours.
  - status.jenkins.io updated
- More communication added:
  - https://github.com/jenkinsci/.github/pull/106
  - https://github.com/jenkins-infra/.github/pull/29

**Comment by dduportal** [comment]
- The operation was finished successfully at 10:30 UTC on JFrog side (confirmed at 15:30 UTC by email).
- The Jenkins infrastructure was set back to normal (controller handling builds) around 14:30 UTC. Confirmed that builds were successfully using the repository around 14:45 UTC.
- It took ~2 hours to ci.jenkins.io to handle the build queue of 974 builds.
